### PR TITLE
CLI tools necessary for the binary targets proposal

### PR DESCRIPTION
**Pull request by hartbit**

### CODE REVIEW

File: Sources/Workspace/Workspace.swift [hunk 1]
```diff
@@ -386,6 +389,7 @@ public class Workspace {
         config: SwiftPMConfig = SwiftPMConfig(),
         fileSystem: FileSystem = localFileSystem,
         repositoryProvider: RepositoryProvider = GitRepositoryProvider(),
+        checksumAlgorithm: HashAlgorithm = SHA256(),
```
Comment (aciidb0mb3r) [inline Sources/Workspace/Workspace.swift]: Can you switch this to SHA256 from CryptoKit if it's available?

Comment (aciidb0mb3r) [reply Sources/Workspace/Workspace.swift]: i.e.:

```
if @available(macOS 10.15, *) {
    return CryptoKit.SHA256(..)
} else {
    return TSCBasic.SHA256(..)
}
```

Comment (hartbit) [reply Sources/Workspace/Workspace.swift]: I've updated the PR to use CryptoKit.SHA256 on macOS 10.15.

**Comment by hartbit** [comment]
@swift-ci smoke test self hosted

**Comment by hartbit** [comment]
@swift-ci please smoke test

**Pull request merged**